src/Sentry/Scope.cs

```diff
@@ -78,11 +82,76 @@ internal SentryId LastEventId
         /// <see cref="Evaluate"/>
         internal event EventHandler? OnEvaluating;
 
+        /// <inheritdoc />
+        [DataMember(Name = "level", EmitDefaultValue = false)]
+        public SentryLevel? Level { get; set; }
+
+        /// <inheritdoc />
+        [DataMember(Name = "transaction", EmitDefaultValue = false)]
+        public string? Transaction { get; set; }
+
+        [DataMember(Name = "request", EmitDefaultValue = false)]
+        private Request? _request;
+
+        /// <inheritdoc />
+        public Request Request
+        {
+            get => _request ??= new Request();
+            set => _request = value;
+        }
+
+        [DataMember(Name = "contexts", EmitDefaultValue = false)]
+        private Contexts? _contexts;
+
+        /// <inheritdoc />
+        public Contexts Contexts
+        {
+            get => _contexts ??= new Contexts();
+            set => _contexts = value;
+        }
+
+        [DataMember(Name = "user", EmitDefaultValue = false)]
+        private User? _user;
+
+        /// <inheritdoc />
+        public User User
+        {
+            get => _user ??= new User();
+            set => _user = value;
+        }
+
+        /// <inheritdoc />
+        [DataMember(Name = "environment", EmitDefaultValue = false)]
+        public string? Environment { get; set; }
+
+        /// <inheritdoc />
+        [DataMember(Name = "sdk", EmitDefaultValue = false)]
+        public SdkVersion? Sdk { get; internal set; } = new SdkVersion();
+
+        /// <inheritdoc />
+        [DataMember(Name = "fingerprint", EmitDefaultValue = false)]
+        [DontSerializeEmpty]
+        public IEnumerable<string> Fingerprint { get; set; } = Enumerable.Empty<string>();
+
+        /// <inheritdoc />
+        [DataMember(Name = "breadcrumbs", EmitDefaultValue = false)]
+        [DontSerializeEmpty]
+        public ICollection<Breadcrumb> Breadcrumbs { get; } = new SynchronizedList<Breadcrumb>();
```

Comment:
I'm not a fan of this custom "synchronized list". Doesnt' ConcurrentBag and List both implement IEnumerable? 

The extension method is internal and can type cast to the type we know we support in both implementation.

Reply:
IEnumerable doesn't have .Add().
ConcurrentBag implements ICollection but doesn't guarantee order.

Reply:
We don't want `Add` exposed to the user. We need `Add` ourselves on the `AddBreadcrumb` method (before we just typecast to the right type)

Reply:
Right about the order. I don't see the reason to change the way this was done before tbqh

Reply:
I wanted to avoid leaky abstraction from `IScope`, because then `AddBreadcrumb` would be coupled to both `SentryEvent` and `Scope` at runtime without a way to validate its assumptions statically.

Reply:
Another approach is to add `AddBreadcrumb` directly to `IScope`, which would eliminate the abstraction issues and would probably be the cleanest solution.

Reply:
Anyway, I don't really care about this too much and I guess it's safer to just keep things how they were. Reverted it back to `ConcurrentQueue` on Scope.

Reply:
OK adding to the interface sounds OK